android/java/org/chromium/chrome/browser/settings/BraveSearchEnginesPreferences.java

```diff
@@ -32,16 +45,76 @@ public void onResume() {
         new Handler().post(() -> updateSearchEnginePreference());
     }
 
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        mManagedPreferenceDelegate = createManagedPreferenceDelegate();
+    }
+
+    private ChromeManagedPreferenceDelegate createManagedPreferenceDelegate() {
+        return new ChromeManagedPreferenceDelegate(getProfile()) {
+            @Override
+            public boolean isPreferenceControlledByPolicy(Preference preference) {
+                String key = preference.getKey();
+                if (PREF_SEARCH_SUGGESTIONS.equals(key)) {
+                    return UserPrefs.get(getProfile())
+                            .isManagedPreference(Pref.SEARCH_SUGGEST_ENABLED);
+                }
+                return false;
+            }
+        };
+    }
+
     private void updateSearchEnginePreference() {
-        Profile lastUsedRegularProfile = ProfileManager.getLastUsedRegularProfile();
         Preference searchEnginePreference = findPreference(PREF_STANDARD_SEARCH_ENGINE);
         searchEnginePreference.setEnabled(true);
         searchEnginePreference.setSummary(
-                BraveSearchEngineUtils.getDSEShortName(lastUsedRegularProfile, false));
+                BraveSearchEngineUtils.getDSEShortName(getProfile(), false));
 
         searchEnginePreference = findPreference(PREF_PRIVATE_SEARCH_ENGINE);
         searchEnginePreference.setEnabled(true);
-        searchEnginePreference.setSummary(BraveSearchEngineUtils.getDSEShortName(
-                lastUsedRegularProfile.getPrimaryOTRProfile(/* createIfNeeded= */ true), true));
+        searchEnginePreference.setSummary(
+                BraveSearchEngineUtils.getDSEShortName(
+                        getProfile().getPrimaryOTRProfile(/* createIfNeeded= */ true), true));
+
+        mSearchSuggestions = (ChromeSwitchPreference) findPreference(PREF_SEARCH_SUGGESTIONS);
+        mSearchSuggestions.setOnPreferenceChangeListener(this);
+        mSearchSuggestions.setManagedPreferenceDelegate(mManagedPreferenceDelegate);
+
+        mShowAutocompleteInAddressBar =
+                (ChromeSwitchPreference) findPreference(PREF_SHOW_AUTOCOMPLETE_IN_ADDRESS_BAR);
+        mShowAutocompleteInAddressBar.setOnPreferenceChangeListener(this);
+
+        mAutocompleteTopSites =
```

Comment:
I'm not sure why `AutocompleteInAddressBar` and `AutocompleteTopSites`  are in `Search engine preference`. They are technically have nothing to do with search engine. Was this move agreed with design team?

Reply:
yes it was agreed upon. we will be following up with more granular details for the option. CC @timchilds @anthonypkeane